### PR TITLE
we work with syms, don't transform to doubles!!

### PR DESCRIPTION
**Pull request by latot**
Cleaning!

### CODE REVIEW

**Comment by cbm755** [comment]
I tend to agree.  But IIRC, Matlab's SMT does this :(  We should check.

**Comment by latot** [comment]
well considering that SMT behavior, i propose use this, isOctave its like isunix, because its little bussy write everywhere ```exist('OCTAVE_VERSION', 'builtin')``` and its more easy to understand.

**Comment by cbm755** [comment]
hmmm, maybe...  I think usually octave tries to be matlab compatible.  Does `@sym/coeffs` do what we want?

The `exist('OCTAVE_VERSION'` thing is recommended upstream.  See also https://github.com/catch22/octave-doctest/blob/master/inst/private/is_octave.m.  Maybe I would consider this but it needs to be its own PR.

**Comment by latot** [comment]
yes, its basically the same, but in isOctave i avoid call it every time, and i try use it in private folder but the problem was the tests, these can't access to the private functions.
While i'll split this.

**Comment by cbm755** [comment]
This is a pretty minor point: we already do what upstream suggests: maybe don't bother.  As you say, it won't work in tests anyway, which is the main place we use this.

**Comment by latot** [comment]
Fixed!

**Comment by cbm755** [comment]
Multiple behaviours is dangerous: need to test both, support both, etc.  Also, running on Matlab is not really being tested anymore (not by me anyway).

So either:

1.  we stay with SMT's rather foolish behaviour (AND SHOUT LOUDLY in our docs---as we already do).
2.  deliberately break compatibility.

I think 1.

**Comment by cbm755** [comment]
Wait, we could add a `sym2polysane`.  Because `coeffs` does not really do the same thing (because it skips zeros).

Then `sym2poly` would just be a wrapper doing `double(sym2polysane(varargin{:}))`...   Thoughts?

**Comment by latot** [comment]
Yay!

**Comment by latot** [comment]
okidokis, Fixed!

File: inst/@sym/sym2polysane.m [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+%% Created: 18 April 2003
+%% Changed: 25 April 2003
+%%    Removed the use of differentiate to get to coefficients - round-off
+%%     errors cause problems. Now using newly created sumterms().
+%% Changed: 6 May 2003
+%%    Removed the attempt to use ldegree(), degree() and coeff() - results
+%%     with these are inconsistent.
+%% Changed: 16 April 2014
+%%    Used the comment header and tests in OctSymPy, but rewrote
+%%    the body (by Colin Macdonald).
```
Comment (latot) [inline inst/@sym/sym2polysane.m]: Can i remove the history?

Comment (cbm755) [inline inst/@sym/sym2polysane.m]: Yes, go for it.

File: inst/@sym/sym2poly.m [hunk 2]
```diff
@@ -30,29 +31,22 @@
 %% @group
 %% syms x y
 %% sym2poly(2*x^2 + 3*x - pi, x)
-%%    @result{} (sym) [2  3  -π]  (1×3 matrix)
+%%    @result{} ans =
+%%   2.0000   3.0000  -3.1416
 %% sym2poly(x^2 + y*x, x)
```
Comment (cbm755) [inline inst/@sym/sym2poly.m]: Do you think we should change the behaviour when x is specified?  I guess that's more consistent behaviour, but its a change from previous Octsympy releases...

Also line 27/28 says we return a row vector of symbolic expressions...

Comment (latot) [inline inst/@sym/sym2poly.m]: okidokis, while i'll open anew issue for this things.

File: inst/@sym/sym2poly.m [hunk 3]
```diff
@@ -115,20 +73,19 @@
...
-%!assert (isequal (sym2poly (x^2+y*x, x), [sym(1) y sym(0)]))
-%!assert (isequal (sym2poly (x^2+y*x, y), [x x^2]))
+%! assert (isequal (sym2poly (x^2+y*x, x), [sym(1) y sym(0)]))
+%! assert (isequal (sym2poly (x^2+y*x, y), [x x^2]))
 %% inverse relationship
-%!assert (isequal (sym2poly (poly2sym ([a b c], x), x), [a b c]))
-%!assert (isequal (poly2sym (sym2poly(a*x^2 + c, x), x), a*x^2 + c))
-%!assert (isequal (sym2poly (poly2sym ([1 2 3])), [1 2 3]))
+%! assert (isequal (sym2poly (poly2sym ([a b c], x), x), [a b c]))
+%! assert (isequal (poly2sym (sym2poly(a*x^2 + c, x), x), a*x^2 + c))
```
Comment (latot) [inline inst/@sym/sym2poly.m]: only to know, what is the meaning change?

**Comment by latot** [comment]
Okidokis, Updated!

File: inst/@sym/sym2poly.m (hunk 3 above)
Comment (cbm755) [inline inst/@sym/sym2poly.m]: > only to know, what is the meaning change?

These are two tests:
````
%!assert ...
%!assert ...
````

These are two assert statements within one test:
````
%!test
%! assert ...
%! assert ...
````

**Comment by latot** [comment]
Code ready and formatted.

**Comment by cbm755** [comment]
I'm adding the new `coeffs(..., 'all')` support in #708.  Perhaps we no longer need `sym2polysane`?

**Comment by latot** [comment]
okis, so will need rewrite this and use reverse in coeffs when is ready, Thx!

**Comment by cbm755** [comment]
I mean, maybe we don't need this at all?

The documentation of `sym2poly` could be improved to point at `coeffs('all')`.

**Comment by latot** [comment]
mm, i would like add that to the doc, and rewrite sym2poly with coeffs function, what do you think? (with that i think we can simplify code)

**Comment by cbm755** [comment]
Sure, that sounds good if it leads to simpler code.

**Pull request closed without merging**